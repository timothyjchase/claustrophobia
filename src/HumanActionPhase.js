import React from 'react'
import { observer } from 'mobx-react'
import { Button, Icon } from 'semantic-ui-react'
import EventMessage from './EventMessage'

const HumanActionPhase = ({ game }) => (
  <div>
    {!!game.event &&
      game.event.phase === 'HUMAN_ACTION' && <EventMessage game={game} />}
    <p>Record removed warriors and the following actions below:</p>
    <Button.Group vertical fluid>
      <Button onClick={() => game.auraOfBlessing()}>
        <Icon name="heart" /> Aura of Blessing
      </Button>
      {!game.oilForYourLamp && (
        <Button onClick={() => game.demonicMechanismTile()}>
          <Icon name="cog" /> Demonic Mechanism tile
        </Button>
      )}
      {game.scenario === 'THE_POSSESSED' && (
        <Button onClick={() => game.addDemon()}>
          <Icon name="box" /> Stash tile (Add Possessed One)
        </Button>
      )}
      {game.scenario === 'HIT_THEM_WHERE_IT_HURTS' &&
        !game.demonsInPlay && (
          <Button onClick={() => game.addDemon()}>
            <Icon name="star outline" /> Pentacle Room tile (Add Master of
            Souls)
          </Button>
        )}
      {game.scenario === 'HIT_THEM_WHERE_IT_HURTS' &&
        game.threatDice >= 3 && (
          <Button onClick={() => game.anyOtherTile()}>
            <Icon name="square plus outline" /> Any other tile (Add 2
            Troglodytes)
          </Button>
        )}
      <Button primary onClick={() => game.completeHumanActionPhase()}>
        <Icon name="play" /> Next Phase
      </Button>
    </Button.Group>
  </div>
)

export default observer(HumanActionPhase)
