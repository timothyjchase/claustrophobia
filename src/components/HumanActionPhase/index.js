import React from 'react'
import PropTypes from 'prop-types'
import { Button } from 'semantic-ui-react'

const HumanActionPhase = ({
  scenarioKey,
  oilForYourLamp,
  demonsInPlay,
  threatDice,
  addDemon,
  placeDemonicMechanismTile,
  placeTile,
  completeHumanActionPhase,
}) => (
  <div>
    <p>Record removed warriors and the following actions below:</p>
    <Button.Group vertical fluid>
      {scenarioKey !== 'THE_RITUAL' && (
        <Button
          disabled={oilForYourLamp}
          icon="cog"
          content="Demonic Mechanism tile (+2 Threat)"
          onClick={placeDemonicMechanismTile}
        />
      )}
      {scenarioKey === 'THE_POSSESSED' && (
        <Button
          icon="box"
          content="Stash tile (+1 Possessed One)"
          onClick={addDemon}
        />
      )}
      {scenarioKey === 'HIT_THEM_WHERE_IT_HURTS' &&
        !demonsInPlay && (
          <Button
            icon="star outline"
            content="Pentacle Room tile (+1 Master of Souls)"
            onClick={addDemon}
          />
        )}
      {scenarioKey === 'HIT_THEM_WHERE_IT_HURTS' && (
        <Button
          disabled={threatDice < 3}
          icon="square plus outline"
          content="Any other tile (+2 Troglodytes)"
          onClick={placeTile}
        />
      )}
      <Button
        primary
        icon="play"
        content="Next Phase"
        onClick={completeHumanActionPhase}
      />
    </Button.Group>
  </div>
)

HumanActionPhase.propTypes = {
  scenarioKey: PropTypes.string.isRequired,
  oilForYourLamp: PropTypes.bool.isRequired,
  demonsInPlay: PropTypes.number.isRequired,
  threatDice: PropTypes.number.isRequired,
  addDemon: PropTypes.func.isRequired,
  placeDemonicMechanismTile: PropTypes.func.isRequired,
  placeTile: PropTypes.func.isRequired,
  completeHumanActionPhase: PropTypes.func.isRequired,
}

export default HumanActionPhase
