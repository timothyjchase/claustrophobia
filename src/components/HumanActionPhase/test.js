import React from 'react'
import HumanActionPhase from './index'

describe('<HumanActionPhase />', () => {
  const commonProps = {
    oilForYourLamp: false,
    demonsInPlay: 0,
    threatDice: 6,
    addDemon: jest.fn(),
    placeDemonicMechanismTile: jest.fn(),
    placeTile: jest.fn(),
    completeHumanActionPhase: jest.fn(),
  }
  describe('with scenario = THE_POSSESSED, oilForYourLamp = true', () => {
    const component = shallow(
      <HumanActionPhase
        {...commonProps}
        scenarioKey="THE_POSSESSED"
        oilForYourLamp
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenario = THE_POSSESSED, oilForYourLamp = false', () => {
    const component = shallow(
      <HumanActionPhase {...commonProps} scenarioKey="THE_POSSESSED" />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenario = THE_RITUAL, oilForYourLamp = false', () => {
    const component = shallow(
      <HumanActionPhase {...commonProps} scenarioKey="THE_RITUAL" />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenario = HIT_THEM_WHERE_IT_HURTS, demonsInPlay = 0, threatDice = 2', () => {
    const component = shallow(
      <HumanActionPhase
        {...commonProps}
        scenarioKey="HIT_THEM_WHERE_IT_HURTS"
        threatDice={2}
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenario = HIT_THEM_WHERE_IT_HURTS, demonsInPlay = 1, threatDice = 3', () => {
    const component = shallow(
      <HumanActionPhase
        {...commonProps}
        scenarioKey="HIT_THEM_WHERE_IT_HURTS"
        demonsInPlay={1}
        threatDice={3}
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
