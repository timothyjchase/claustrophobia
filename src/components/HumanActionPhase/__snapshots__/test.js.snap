// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HumanActionPhase /> with scenario = HIT_THEM_WHERE_IT_HURTS, demonsInPlay = 0, threatDice = 2 should match snapshot 1`] = `
<div>
  <p>
    Record removed warriors and the following actions below:
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Demonic Mechanism tile (+2 Threat)"
      disabled={false}
      icon="cog"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Pentacle Room tile (+1 Master of Souls)"
      icon="star outline"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Any other tile (+2 Troglodytes)"
      disabled={true}
      icon="square plus outline"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<HumanActionPhase /> with scenario = HIT_THEM_WHERE_IT_HURTS, demonsInPlay = 1, threatDice = 3 should match snapshot 1`] = `
<div>
  <p>
    Record removed warriors and the following actions below:
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Demonic Mechanism tile (+2 Threat)"
      disabled={false}
      icon="cog"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Any other tile (+2 Troglodytes)"
      disabled={false}
      icon="square plus outline"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<HumanActionPhase /> with scenario = THE_POSSESSED, oilForYourLamp = false should match snapshot 1`] = `
<div>
  <p>
    Record removed warriors and the following actions below:
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Demonic Mechanism tile (+2 Threat)"
      disabled={false}
      icon="cog"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Stash tile (+1 Possessed One)"
      icon="box"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<HumanActionPhase /> with scenario = THE_POSSESSED, oilForYourLamp = true should match snapshot 1`] = `
<div>
  <p>
    Record removed warriors and the following actions below:
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Demonic Mechanism tile (+2 Threat)"
      disabled={true}
      icon="cog"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Stash tile (+1 Possessed One)"
      icon="box"
      onClick={[MockFunction]}
      role="button"
    />
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;

exports[`<HumanActionPhase /> with scenario = THE_RITUAL, oilForYourLamp = false should match snapshot 1`] = `
<div>
  <p>
    Record removed warriors and the following actions below:
  </p>
  <ButtonGroup
    fluid={true}
    vertical={true}
  >
    <Button
      as="button"
      content="Next Phase"
      icon="play"
      onClick={[MockFunction]}
      primary={true}
      role="button"
    />
  </ButtonGroup>
</div>
`;
